File: Templates/Day_2/WorkshareSchedule/C/gauss2d_serial.c
#include <stdio.h>
#include <math.h>

// parameters N and h
#define NSIZE 44800
#define STEPSIZE 0.0002

int main()
{

  double xpos, ypos;
  double integral=0.0;
  int numThreads;


  for ( int i = 0; i < NSIZE; i++)
    {
      xpos = (double) i * STEPSIZE;
      for (int j = i+1; j < NSIZE; j++ )
	{
	  ypos = (double) j * STEPSIZE;
	  double xvecSquare = xpos*xpos + ypos*ypos;
	  integral += exp(-xvecSquare) * STEPSIZE * STEPSIZE;
	}  // j-loop
    }  // i-loop
    
  integral *= 8.0;

  printf("Approximate integral: %f for n: %i and stepsize: %f \n",
	 integral, NSIZE, STEPSIZE);

  return 0;

}

File: Templates/Day_2/WorkshareSchedule/Fortran/gauss2d_serial.f90
Program gauss2d

  implicit none

  ! parameters N and h
  integer, parameter :: nsize    = 44800
  double precision, parameter :: stepsize = 0.0002D0

  
  double precision :: xpos, ypos
  double precision :: integral

  ! summation indices
  integer :: i, j

  integral = 0.0D0
  
  do i = 0, nsize-1
     xpos = i * stepsize
     do j = i+1, nsize-1
        ypos = j * stepsize
        integral = integral + &
             stepsize*stepsize*exp(-xpos*xpos - ypos*ypos)
     enddo  ! j-loop
  enddo     !i-loop
  
  integral = 8.0D0 * integral

  print *, "Approximate integral:", integral, " for n:", nsize, " and stepsize:", stepsize 

End program gauss2d
